Tests/ArgumentParserUnitTests/HelpGenerationTests+GroupName.swift

```diff
@@ -187,18 +187,23 @@ extension HelpGenerationTests {
         """)
   }
 
+  fileprivate struct NestedGroups: ParsableArguments {
+    @OptionGroup(visibility: .hidden)
+    var flagsAndOptions: FlagsAndOptions
+
+    @OptionGroup(visibility: .private)
+    var argsAndFlags: ArgsAndFlags
+  }
+
   fileprivate struct HiddenGroups: ParsableCommand {
     @OptionGroup(title: "Flags Group", visibility: .hidden)
     var flags: Flags
 
     @OptionGroup(title: "Options Group", visibility: .hidden)
     var options: Options
 
-    @OptionGroup(visibility: .hidden)
-    var flagsAndOptions: FlagsAndOptions
-
-    @OptionGroup(visibility: .private)
-    var argsAndFlags: ArgsAndFlags
+    @OptionGroup()
+    var nested: NestedGroups
```

Comment:
I think for completeness it would be best to keep this existing type and add a new `NestedHiddenGroups`/`testNestedHiddenGroups()` type/test function pair.